drivers/nrf_802154_sl_opensource/src/platform/gpiote/nrf_802154_gpiote_zephyr.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ *   This file contains implementation of the nRF 802.15.4 GPIOTE abstraction.
+ *
+ * This implementation is to be used with Zephyr RTOS.
+ *
+ */
+
+#include <platform/gpiote/nrf_802154_gpiote.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <sys/__assert.h>
+#include <device.h>
+#include <toolchain.h>
+#include <drivers/gpio.h>
+
+#include <hal/nrf_gpio.h>
+#include <nrf_802154_sl_coex.h>
+#include <nrf_802154_sl_utils.h>
+
+const static struct device *dev;
+static struct gpio_callback grant_cb;
+static uint32_t pin_number = COEX_GPIO_PIN_INVALID;
+
+static void gpiote_irq_handler(const struct device *gpiob, struct gpio_callback *cb,
+			       uint32_t pins)
+{
+	ARG_UNUSED(gpiob);
+	ARG_UNUSED(cb);
+	ARG_UNUSED(pins);
+
+	nrf_802154_wifi_coex_gpiote_irqhandler();
+}
+
+void nrf_802154_gpiote_init(void)
+{
+	switch (nrf_802154_wifi_coex_interface_type_id_get()) {
+	case NRF_802154_WIFI_COEX_IF_NONE:
+		return;
+
+	case NRF_802154_WIFI_COEX_IF_3WIRE: {
+		nrf_802154_wifi_coex_3wire_if_config_t cfg;
+
+		nrf_802154_wifi_coex_cfg_3wire_get(&cfg);
+
+		pin_number = cfg.grant_cfg.gpio_pin;
+		__ASSERT_NO_MSG(pin_number != COEX_GPIO_PIN_INVALID);
+
+		bool use_port_1 = (pin_number > P0_PIN_NUM);
+
+		/* Convert to the Zephyr primitive */
+		pin_number = use_port_1 ? pin_number - P0_PIN_NUM : pin_number;
```

Comment:
This assumes that `P0_PIN_NUM` is 32, which not necessarily will be always true.
You can use `nrf_gpio_pin_port_number_extract()` instead.